### ISSUE TITLE
Pip install doesn't work

### ISSUE THREAD

**Issue by mawright** [a-r-j/graphein#273]
**Describe the bug**
The straightforward pip install described in the installation directions fails.

**To Reproduce**

1. `pip install graphein`
2. Pip fails

**Expected behavior**
Graphein to install successfully

**Screenshots**
```
$ conda create -n test -qy pip && conda activate test && pip install graphein
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/matt/miniconda3/envs/test

  added / updated specs:
    - pip


The following NEW packages will be INSTALLED:

  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main 
  _openmp_mutex      pkgs/main/linux-64::_openmp_mutex-5.1-1_gnu 
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0 
  ca-certificates    pkgs/main/linux-64::ca-certificates-2023.01.10-h06a4308_0 
  certifi            pkgs/main/linux-64::certifi-2022.12.7-py310h06a4308_0 
  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.38-h1181459_1 
  libffi             pkgs/main/linux-64::libffi-3.4.2-h6a678d5_6 
  libgcc-ng          pkgs/main/linux-64::libgcc-ng-11.2.0-h1234567_1 
  libgomp            pkgs/main/linux-64::libgomp-11.2.0-h1234567_1 
  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-11.2.0-h1234567_1 
  libuuid            pkgs/main/linux-64::libuuid-1.41.5-h5eee18b_0 
  ncurses            pkgs/main/linux-64::ncurses-6.4-h6a678d5_0 
  openssl            pkgs/main/linux-64::openssl-1.1.1t-h7f8727e_0 
  pip                pkgs/main/linux-64::pip-22.3.1-py310h06a4308_0 
  python             pkgs/main/linux-64::python-3.10.9-h7a1cb2a_1 
  readline           pkgs/main/linux-64::readline-8.2-h5eee18b_0 
  setuptools         pkgs/main/linux-64::setuptools-65.6.3-py310h06a4308_0 
  sqlite             pkgs/main/linux-64::sqlite-3.40.1-h5082296_0 
  tk                 pkgs/main/linux-64::tk-8.6.12-h1ccaba5_0 
  tzdata             pkgs/main/noarch::tzdata-2022g-h04d1e81_0 
  wheel              pkgs/main/linux-64::wheel-0.38.4-py310h06a4308_0 
  xz                 pkgs/main/linux-64::xz-5.2.10-h5eee18b_1 
  zlib               pkgs/main/linux-64::zlib-1.2.13-h5eee18b_0 


Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Collecting graphein
  Using cached graphein-1.5.2.tar.gz (185 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [45 lines of output]
      {'all': {'black',
               'flake8',
               'hypothesis',
               'interrogate',
               'isort',
               'nbstripout',
               'nbval',
               'pandoc',
               'pre-commit',
               'pycodestyle',
               'pydocstyle',
               'pytest',
               'pytest-cov',
               'pytest-xdist'},
       'dev': ['black',
               'flake8',
               'hypothesis',
               'interrogate',
               'isort',
               'nbstripout',
               'nbval',
               'pandoc',
               'pre-commit',
               'pycodestyle',
               'pydocstyle',
               'pytest',
               'pytest-cov',
               'pytest-xdist'],
       'extras': ['black',
                  'flake8',
                  'hypothesis',
                  'interrogate',
                  'isort',
                  'nbstripout',
                  'nbval',
                  'pandoc',
                  'pre-commit',
                  'pycodestyle',
                  'pydocstyle',
                  'pytest',
                  'pytest-cov',
                  'pytest-xdist']}
      error in graphein setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Expected end or semicolon (after version specifier)
          pyyaml>=5.1,<6.*
                ~~~~~~~~^
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

```

**Desktop (please complete the following information):**

- OS: Ubuntu 22.04
- Python Version: I tried this with fresh conda environments of both 3.8 and 3.10.
- Graphein Version [e.g. 22]: pip head as of March 2, 2023

**Comment by a-r-j** [comment]
Thanks for the bug report. Seems like setuptools has dropped support for wildcard dependency versions. This has been resolved on master. You can install Graphein from GitHub pending a PyPI release with a fix.

**Comment by a-r-j** [comment]
Now resolved in 1.6.0

`pip install graphein=1.6.0`
